catkin_tools/jobs/catkin.py

```diff
@@ -102,6 +130,7 @@ def create_catkin_build_job(context, package, package_path, dependencies, force_
         'make',
         [MAKE_EXEC] + make_args,
         cwd=build_space,
+        env=ctr_env if 'test' in make_args else {},
```

Comment:
Yeah, that's the behavior of the `CommandStage` see: https://github.com/catkin/catkin_tools/pull/249/files#diff-62ab2c322956d57c062b8f2b885c4c9cR42